router/usecase/quote_test.go

```diff
@@ -122,127 +123,334 @@ func (s *RouterTestSuite) TestPrepareResult() {
+						RouteImpl: route.RouteImpl{
+							Pools: []domain.RoutablePool{
+								s.newRoutablePool(
+									sqsdomain.NewPool(poolThree, poolThree.GetSpreadFactor(sdk.Context{}), poolThreeBalances),
+									USDC,
+									takerFeeThree,
+									domain.CosmWasmPoolRouterConfig{},
+								),
+							},
+						},
+
+						InAmount:  totalInAmount.QuoRaw(2),
+						OutAmount: totalOutAmount.QuoRaw(2),
 					},
 				},
-
-				InAmount:  totalInAmount.QuoRaw(2),
-				OutAmount: totalOutAmount.QuoRaw(2),
+				EffectiveFee: osmomath.ZeroDec(),
 			},
-
-			// Route 2
-			&usecase.RouteWithOutAmount{
-				RouteImpl: route.RouteImpl{
-					Pools: []domain.RoutablePool{
-						s.newRoutablePool(
-							sqsdomain.NewPool(poolThree, poolThree.GetSpreadFactor(sdk.Context{}), poolThreeBalances),
-							USDC,
-							takerFeeThree,
-							domain.CosmWasmPoolRouterConfig{},
-						),
+			expectedRoutes: []domain.SplitRoute{
+				// Route 1
+				&usecase.RouteWithOutAmount{
+					RouteImpl: route.RouteImpl{
+						Pools: []domain.RoutablePool{
+							pools.NewRoutableResultPool(
+								poolIDOne,
+								poolmanagertypes.Balancer,
+								poolOne.GetSpreadFactor(sdk.Context{}),
+								USDT,
+								takerFeeOne,
+								notCosmWasmPoolCodeID,
+							),
+							pools.NewRoutableResultPool(
+								poolIDTwo,
+								poolmanagertypes.Balancer,
+								poolTwo.GetSpreadFactor(sdk.Context{}),
+								USDC,
+								takerFeeTwo,
+								notCosmWasmPoolCodeID,
+							),
+						},
 					},
+
+					InAmount:  totalInAmount.QuoRaw(2),
+					OutAmount: totalOutAmount.QuoRaw(2),
 				},
 
-				InAmount:  totalInAmount.QuoRaw(2),
-				OutAmount: totalOutAmount.QuoRaw(2),
-			},
-		},
-		EffectiveFee: osmomath.ZeroDec(),
-	}
+				// Route 2
+				&usecase.RouteWithOutAmount{
+					RouteImpl: route.RouteImpl{
+						Pools: []domain.RoutablePool{
+							pools.NewRoutableResultPool(
+								poolIDThree,
+								poolmanagertypes.Balancer,
+								poolThree.GetSpreadFactor(sdk.Context{}),
+								USDC,
+								takerFeeThree,
+								notCosmWasmPoolCodeID,
+							),
+						},
+					},
 
-	expectedRoutes := []domain.SplitRoute{
-
-		// Route 1
-		&usecase.RouteWithOutAmount{
-			RouteImpl: route.RouteImpl{
-				Pools: []domain.RoutablePool{
-					pools.NewRoutableResultPool(
-						poolIDOne,
-						poolmanagertypes.Balancer,
-						poolOne.GetSpreadFactor(sdk.Context{}),
-						USDT,
-						takerFeeOne,
-						notCosmWasmPoolCodeID,
-					),
-					pools.NewRoutableResultPool(
-						poolIDTwo,
-						poolmanagertypes.Balancer,
-						poolTwo.GetSpreadFactor(sdk.Context{}),
-						USDC,
-						takerFeeTwo,
-						notCosmWasmPoolCodeID,
-					),
+					InAmount:  totalInAmount.QuoRaw(2),
+					OutAmount: totalOutAmount.QuoRaw(2),
 				},
 			},
-
-			InAmount:  totalInAmount.QuoRaw(2),
-			OutAmount: totalOutAmount.QuoRaw(2),
+			// (0.02 + (1 - 0.02) * 0.0004) * 0.5 + 0.003 * 0.5
+			expectedEffectiveFee: "0.011696000000000000",
+			expectedJSON: `{
```

Comment:
Let's remove the `expectedJSON` because it is more difficult to maintain. Every time we make a change, we would have to modify these payloads which would be time-consuming.

Instead, we can assert similar guarantees more dynamically at the integration test level

Reply:
For a reference:  This suggestion was accepted since [it is](https://github.com/osmosis-labs/sqs/pull/397/files#diff-997a6364dda024fd4d1a647f5de61deb04cd144f434ea04a0d6839dc68af9406R176) using same approach and test files as [handlers](https://github.com/osmosis-labs/sqs/pull/397/files#diff-847392b583119d093d0335703a7502a31c338dfac440da34d427cece264d0cf7R96).